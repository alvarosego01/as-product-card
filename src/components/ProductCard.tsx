

import { useProducts } from "../hooks/useProducts";
import React, { createContext } from "react";
import { InitialValues_I, ProductCardHandlers_I, ProductContextProps_I, Product_I, onChangeArgs_I } from "../interfaces/interfaces";

import styles from "../styles/styles.module.css";


export const ProductContext = createContext({

} as ProductContextProps_I)
const { Provider } = ProductContext;


export interface Props_I {
    product: Product_I;
    // children?: ReactElement | ReactElement[];
    children: (args: ProductCardHandlers_I) => JSX.Element;
    className?: string;
    style?: React.CSSProperties;
    onChange?: (args: onChangeArgs_I) => void;
    value?: number;
    initialValues?: InitialValues_I
}


export const ProductCard = ({ children, product, className, style, onChange, value, initialValues }: Props_I) => {

    const {
        counter,
        increaseBy,
        maxCount,
        isMaxCountReached,
        reset
    } = useProducts({
        onChange,
        product,
        value,
        initialValues
    });

    return (
        // Se provee la informaciÂ´pn que necesitan los hijos
        <Provider value={{
            counter,
            increaseBy,
            product,
            maxCount,
        }}>
            <div style={style} className={`${styles.productCard} ${className}`}>
                {
                    children({
                        count: counter,
                        isMaxCountReached,
                        maxCount,
                        product,
                        increaseBy,
                        reset
                    })
                }
            </div>
        </Provider>
    )

}


