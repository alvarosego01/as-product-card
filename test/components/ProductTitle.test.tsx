import React from 'react';
import ProductCard, { ProductTitle } from '../../src/components';
// import { render } from 'react-dom';
// import { render, screen } from "@testing-library/react";

import renderer from 'react-test-renderer';


import { product1 } from '../data/products';

describe('Pruebas en productTitle', () => {

    test('Debe mostrar component con titulo personalizado', () => {

        const wrapper = renderer.create(
            <ProductTitle title='hola mundo' />
        );

        // console.log('wrapper', wrapper.toJSON());

        expect(wrapper.toJSON()).toMatchSnapshot();


    })

    test('Debe mostrar component con titulo por defecto', () => {

        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {
                    () => (
                        <ProductTitle />
                    )
                }
            </ProductCard>
        );

        // console.log('wrapper', wrapper.toJSON());

        expect(wrapper.toJSON()).toMatchSnapshot();

    })

});
