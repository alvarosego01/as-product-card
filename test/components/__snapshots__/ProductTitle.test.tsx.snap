// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en productTitle Debe mostrar component con titulo personalizado 1`] = `
<span
  className="productDescription undefined"
>
  hola mundo
</span>
`;

exports[`Pruebas en productTitle Debe mostrar component con titulo por defecto 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Titulo 1
  </span>
</div>
`;
